diff --git a/target/linux/rockchip/dts/rk3568/rk3568-t68m.dts b/target/linux/rockchip/dts/rk3568/rk3568-t68m.dts
--- a/target/linux/rockchip/dts/rk3568/rk3568-t68m.dts
+++ b/target/linux/rockchip/dts/rk3568/rk3568-t68m.dts
@@ -211,7 +211,34 @@
 &combphy2 {
 	status = "okay";
 };
 
+&sata2 {
+\tstatus = "okay";
+};
+
+sdio_pwrseq: sdio-pwrseq {
+\tcompatible = "mmc-pwrseq-simple";
+\tclocks = <&rk809 1>;
+\tclock-names = "ext_clock";
+\tpinctrl-names = "default";
+\tpinctrl-0 = <&wifi_enable_h>;
+\treset-gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_LOW>;
+};
+
+&sdmmc2 {
+\tbus-width = <4>;
+\tcap-sd-highspeed;
+\tcap-sdio-irq;
+\tdisable-wp;
+\tkeep-power-in-suspend;
+\tmmc-pwrseq = <&sdio_pwrseq>;
+\tnon-removable;
+\tpinctrl-names = "default";
+\tpinctrl-0 = <&sdmmc2m0_bus4 &sdmmc2m0_clk &sdmmc2m0_cmd>;
+\tsd-uhs-sdr104;
+\tsupports-sdio;
+\tstatus = "okay";
+};
 &cpu0 {
 	cpu-supply = <&vdd_cpu>;
 };
@@ -625,11 +625,17 @@
 	pmic {
 		pmic_int: pmic_int {
 			rockchip,pins = <0 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
 
+\tsdio-pwrseq {
+\t\twifi_enable_h: wifi-enable-h {
+\t\t\trockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_up>;
+\t\t};
+\t};
+
 	rockchip-key {
 		reset_button_pin: reset-button-pin {
 			rockchip,pins = <3 RK_PB1 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
